_health-care/_js/components/form-elements/ErrorableSelect.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import _ from 'lodash';
+
+/**
+ * A form select with a label that can display error messages.
+ *
+ * Validation has the following props.
+ * `errorMessage` - Error string to display in the component.
+ *                  When defined, indicates select has a validation error.
+ * `label` - String for the select field label.
+ * `options` - Array of options to populate select.
+ * `required` - boolean. Render marker indicating field is required.
+ * `value` - string. Value of the select field.
+ * `onUserInput` - a function with this prototype: (newValue)
+ */
+class ErrorableSelect extends React.Component {
+  constructor() {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentWillMount() {
+    this.selectId = _.uniqueId('errorable-select-');
+  }
+
+  handleChange(domEvent) {
+    this.props.onUserInput(domEvent.target.value);
+  }
+
+  render() {
+    // Calculate error state.
+    let errorSpan = '';
+    let errorSpanId = undefined;
+    if (this.props.errorMessage) {
+      errorSpanId = `${this.selectId}-error-message`;
+      errorSpan = <span className="usa-input-error-message" id={`${errorSpanId}`}>{this.props.errorMessage}</span>;
+    }
+
+    // Calculate required.
+    let requiredSpan = '';
+    if (this.props.required) {
+      requiredSpan = <span className="usa-additional_text">Required</span>;
+    }
+
+    // Calculate options for select
+    const options = _.map(this.props.options, (val, key) => {
+      return (
+        <option
+            key={key}
+            value={key}>
+          {val}
+        </option>
+      );
+    });
+    // let reactKey = 0;
+    // const optionElements = this.props.options.map((obj) => {
```

Comment:
The reason yours and mine diverges is I expect the options prop to be an array of objects where each object has a `label` and a `value` property.

In your API, you are expecting an associative array.